core/mixins.py

```diff
@@ -45,6 +46,29 @@ def get_context_data(self, *args, **kwargs):
         return super().get_context_data(page=self.page, *args, **kwargs)
 
 
+class GetCMSComponentMixin:
+    @cached_property
+    def cms_component(self):
+        response = cms_api_client.lookup_by_slug(
```

Comment:
will we only have a max of 1 component per page?

Reply:
for the moment yes. if we add more we could return a dictionary of named components instead of just one. i don't know how likely that would be though